src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,165 @@
 package core.basesyntax;
 
 import java.util.List;
+import java.util.Objects;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> first;
+    private Node<T> last;
+
+    public static class Node<T> {
+        private T value;
+        private Node<T> prev;
+        private Node<T> next;
+
+        public Node(T value, Node<T> prev, Node<T> next) {
+            this.value = value;
+            this.prev = prev;
+            this.next = next;
+        }
+    }
+
     @Override
     public void add(T value) {
+        Node<T> newNode = new Node<>(value, last, null);
+        if (size == 0) {
+            first = newNode;
+        } else {
+            last.next = newNode;
+        }
+        last = newNode;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index < 0 || index > size) {
+            throw new IndexOutOfBoundsException("index < 0");
+        }
+        if (size == 0 || index == size) {
+            addTail(value);
+        } else if (index == 0) {
+            addHead(value);
+        } else {
+            Node<T> newNode = new Node<>(value,
+                    getNodePerIndex(index).prev, getNodePerIndex(index));
+            getNodePerIndex(index).prev.next = newNode;
+            getNodePerIndex(index).prev = newNode;
+            size++;
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T listValue: list) {
+            addTail(listValue);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        if (index >= size || index < 0) {
+            throw new IndexOutOfBoundsException("Exception");
+        }
+        return getNodePerIndex(index).value;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        if (index < 0 || index >= size) {
+            throw new IndexOutOfBoundsException("Not valide index");
+        }
+        T oldValue = getNodePerIndex(index).value;
```

Comment:
same here, let's save the result of `getNodePerIndex(index)` to the local variable